### ISSUE TITLE
Highlighting breaks for some regex inside if condition

### ISSUE THREAD

**Issue by chigur** [JohnNilsson/awk-sublime#7]
First of all thank you for such a great plugin. It really helped me while learning awk. Following is a snippet with parts of the syntax that are highlighted correctly and parts that aren't. I tried debugging the exact problem but I don't know the how the sublime syntax files work.

```awk
# works
/\{([^\}]+)\}/

# works
match($0, /\{([^\}]+)\}/) {}

1 {
  # doesn't work
  if (/\{([^\}]+)\}/) {
  }

  # only regex highlighting works
  if (/\{([^\}]+)\}/) {
  }
}

1 {
  # works by removing `([^\}]+)`
  if (/\{\}/) {
  } else {
  }

  # works by removing `(` and `)`
  if (/\{[^\}]+\}/) {
  } else {
  }
}
```

**Comment by JohnNilsson** [comment]
Thank you! I'm glad it helped you.

Ironically it seems to work in the GitHub highlighter ;) ( see #6  )

I'll take a look at it

**Comment by JohnNilsson** [comment]
I believe this is fixed by 792d9215315758c505a44a325b910bedc87bd10b

**Comment by chigur** [comment]
Thanks. I'll check it out.
